docs/operations/asset-repository.md

```diff
@@ -0,0 +1,58 @@
+# Using local asset repositories
+
+You can configure kOps to provision a cluster to download assets (images and files) from local repositories.
+This is useful when downloading assets from the internet is undersirable, for example:
+
+* To deploy where the network is offline or internet-restricted.
+* To avoid rate limits or network transfer costs.
+* To limit exposure to watering-hole attacks.
+* To comply with other security requirements, such as the need to scan for vulnerabilities.
+
+There can be one repository for images and another for files.
+
+## Configuring
+
+### Configuring a local image repository
+
+To configure a local image repository, set either `assets.containerRegistry` or `assets.containerProxy` in the cluster spec.
+They both do essentially the same thing, but differ in how they calculate the names of the local images.
+
+```yaml
+spec:
+  assets:
+    containerRegistry: example.com/registry
+```
+
+or
+
+```yaml
+spec:
+  assets:
+    containerProxy: example.com/proxy
+```
+
+### Configuring a local file repository
+
+To configure a local file repository, set `assets.fileRepository` in the cluster spec.
+
+```yaml
+spec:
+  assets:
+    fileRepository: https://example.com/files
+```
+
+## Copying
+
+You can copy assets into their repositories either by running `kops update cluster` or through an external process.
+
+When doing a `kops update cluster`, kOps will copy assets into their respective repositories if
+they do not already exist there.
+
+For file assets, kOps only supports copying to a repository that is either an S3 or GCS bucket.
+An S3 bucket must be configured using the [regional naming conventions of S3](https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region).
+A GCS bucket must be configured with a prefix of `https://storage.googleapis.com/`.
```

Comment:
I'm not quite sure why `fileRepository` doesn't take VFS paths. Perhaps a future PR.

Reply:
I think it doesn't accept a VFS path because it is used for NodeUp download too, and that is done via wget.

Reply:
It could convert from VFS form to https form for use by nodeup/userdata. That's a lot more natural than going the other way.

Reply:
It could, but would have to be checked for access and maybe provide some override in case deduced http location is incorrect.